src/parquet/file/metadata.h

```diff
@@ -91,6 +91,32 @@ class PARQUET_EXPORT RowGroupMetaData {
 
 class FileMetaDataBuilder;
 
+struct PARQUET_EXPORT FileVersion {
+  /// Application that wrote the file. e.g. "IMPALA"
+  std::string application;
+
+  /// Version of the application that wrote the file, expressed in three parts
+  /// (<major>.<minor>.<patch>). Unspecified parts default to 0, and extra parts are
+  /// ignored. e.g.:
+  /// "1.2.3"    => {1, 2, 3}
+  /// "1.2"      => {1, 2, 0}
+  /// "1.2-cdh5" => {1, 2, 0}
+  struct {
+    int major;
+    int minor;
+    int patch;
+  } version;
+
+  FileVersion() {}
+  explicit FileVersion(const std::string& created_by);
+
+  /// Returns true if version is strictly less than <major>.<minor>.<patch>
+  bool VersionLt(int major, int minor = 0, int patch = 0) const;
```

Comment:
I'm still not quite following. Can you look at the version check in this patch and let me know what you're proposing changing the code to? We have to parse the application (here, parquet-mr) out of the version string. Having to construct a dummy FileVersion is a bit awkward. I would suggest we merge this patch and you can make changes to suit PARQUET-793